File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.java
class Solution {
    public String solution(int[] numLog) {
        String answer = "";
        
        for (int i = 0; i < numLog.length - 1; i++) {
            int j = numLog[i + 1] - numLog[i];
            
            switch (j) {
                    case 1 -> answer += "w";
                    case -1 -> answer += "s";
                    case 10 -> answer += "d";
                    case -10 -> answer += "a";
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.py
def solution(numLog):
    answer = ''
    
    dict = {1: 'w', -1: 's', 10: 'd', -10: 'a'}
    
    for i in range(len(numLog)):
        try:
            answer += dict[numLog[i+1] - numLog[i]]
        except:
            break
    
    return answer